File: src/phoenix/server/api/input_types/TimeRange.py
from datetime import datetime, timezone
from typing import Optional

import strawberry


@strawberry.input
class TimeRange:
    start: Optional[datetime] = strawberry.field(
        default=None,
        description="The start of the time range",
    )
    end: Optional[datetime] = strawberry.field(
        default=None,
        description="The end of the time range. Right exclusive.",
    )

    def __post_init__(self) -> None:
        if self.start:
            self.start = self.start.astimezone(timezone.utc)
        if self.end:
            self.end = self.end.astimezone(timezone.utc)

    def is_valid(self) -> bool:
        return not self.start or not self.end or self.start < self.end

File: scripts/mock-llm-server/dashboard/src/utils/timeRange.ts
export type TimeRange = "5m" | "30m" | "60m";

export function getTimeRangeMs(range: TimeRange): number {
  switch (range) {
    case "5m":
      return 5 * 60 * 1000;
    case "30m":
      return 30 * 60 * 1000;
    case "60m":
      return 60 * 60 * 1000;
  }
}
